tests/l2/test_l2_configure.py

```diff
@@ -0,0 +1,91 @@
+"""
+Tests related to L2 configuration
+"""
+import logging
+import pytest
+import time
+
+from tests.common import reboot
+from tests.common.helpers.assertions import pytest_expect
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('any')
+]
+
+
+def test_l2_configure(duthosts, rand_one_dut_hostname):
+    """
+    @summary: Test configuring dut as a L2 switch.
+
+    Args:
+        duthosts: set of DUTs.
+    """
+    # Setup.
+    duthost = duthosts[rand_one_dut_hostname]
+    hwsku = duthost.facts["hwsku"]
+
+    # Store original config for comparison.
+    orig_vlan = duthost.shell("show vlan config")["stdout"]
+    orig_int = duthost.shell("show int status")["stdout"]
+
+    # Perform L2 configuration
+    logger.info("Configuring DUT into L2 mode.")
+    l2_cfg = "sudo sonic-cfggen --preset l2 -p -H -k {}" \
+        " | sudo config load /dev/stdin -y".format(hwsku)
+    duthost.shell(l2_cfg)
+    duthost.shell("sudo config qos reload --no-dynamic-buffer")
+    time.sleep(30)
+
+    logger.info("Checking vlan and interface facts.")
+    new_vlan = duthost.shell("show vlan config")["stdout"]
+    new_int = duthost.shell("show int status")["stdout"]
+    pytest_expect(orig_vlan != new_vlan, "vlan config not updated.")
```

Comment:
>orig_vlan != new_vlan [](http://example.com/codeflow?start=18&length=21)

In L2, is there any specific vlan changed? Same to int, also I saw wiki mentioned QoS reload, are you going to check buffer related config as well?